### ISSUE TITLE
Behaviour of degenerate aspect-ratios.

### ISSUE THREAD

**Issue by bfgeek** [w3c/svgwg#848]
In https://svgwg.org/svg-next/single-page.html#coords-SizingSVGInCSS
https://drafts.csswg.org/css-values-4/#degenerate-ratio

The spec talks about when to fallback to using the viewBox for the aspect-ratio. However we don't have interop when the width/height attributes specify a degenerate aspect-ratio. E.g.

```html
<!DOCTYPE html>
<img style="width: 100px; border: solid; background: lime;" src="
    data:image/svg+xml,<svg xmlns='http://www.w3.org/2000/svg' width='0px' height='50px' viewBox='0 0 1 1'></svg>">
```

What should happen in this case?

FF currently does:
  intrinsic-size: 0x50, aspect-ratio: null
Blink/WebKit
  intrinsic-size: 0x50, aspect-ratio: 1/1

Currently with the new CSS `aspect-ratio` property for degenerate aspect-ratios, e.g:
`aspect-ratio: 0/50` will fallback to the intrinsic aspect-ratio if specified.

I think as per the spec at the moment, FF is kinda correct, but the spec doesn't specify what should happen with degenerate aspect-ratios. Blink/WebKit matches the "fallback if degenerate" precedent.

**Comment by bfgeek** [comment]
@fantasai @AmeliaBR @tabatkins Thoughts?

**Comment by bfgeek** [comment]
We can also create most interesting cases like:
```html
<!DOCTYPE html>
<img style="width: 100px; border: solid; background: lime;" src="
    data:image/svg+xml,<svg xmlns='http://www.w3.org/2000/svg' width='-50px' height='50px' viewBox='0 0 1 1'></svg>">
```

**Comment by Doktorchen** [comment]
According to SVG 1.1 recommendation negative values are in error, resulting in error processing.
Zero disables rendering, no attribute results in 100%.
The SVG 2 draft has to meet this behaviour to be backwards compatible and to avoid annoyance for authors and audience.

If SVG 2 still insists, that the CSS  properties width and height apply/overwrite the corresponding attributes, their values simply replace the attribute values before the old rules from 1.1 and 1.2 apply.

Comparable to the declarative animation case switching between 0 and another value is no problem, well defined, it only switches on and off rendering.

This simply means for the first example for presentation:
width 100px, height 50px, viewBox 0 0 1 1 results only in a rescaling as usual with the default
preserveAspectRatio xMidYMid meet.

For the second due to a negative value a presentation program starts with error processing, but after applying the CSS property with a proper value it swiches back to rendering.

**Comment by tabatkins** [comment]
@Doktorchen I think you're misreading the first example; the SVG there has a `width="0px" height="50px"`. The width on the `<img>` is irrelevant here, as it doesn't override or interact with the SVG file. (It just lets us see what the SVG's final aspect ratio is, by observing the height of the `<img>`.)

**Comment by Doktorchen** [comment]
Yes, you are right, something did me let 'see' the style attribute on the svg element ;o)

What remains: 0 width or height disables rendering.
This is a feature in SVG 1.1, especially for animation, a few authors may use it to switch on and off rendering.
But if not rendered, it has no aspect ratio at all.
Might be the same as img referencing an empty 0 Byte file.

Instead of error processing for the second example of course SVG 2 could define some other behaviour - but why, usually authors have to take attention to this before publication anyway, therefore some error processing is pretty useful.

**Comment by bfgeek** [comment]
Sorry one thing to note is that I used the svg-next specification just because that's what I was looking at - but same text in SVG1.1 (https://www.w3.org/TR/SVG11/coords.html) which has similar ambiguities on how to handle this degenerate case. Less interested in the rendering of the SVG, rather the interpretation of the aspect-ratio/intrinsic-size into "CSS embedder".
